### DISCUSSION TITLE
Getting a 1bpp bmp from Image<TPixel> (essentially monochrome bmp)

### DISCUSSION THREAD

**Discussion by kevinstampe** [SixLabors/ImageSharp#1621] [Q&A] [answered]
Hi there, fellow JimBobSquarePants fans! (and ofc James himself👍)

I can't seem to figure out how to get a monochrome bitmap out of ImageSharp.
I need to render a bitmap for a monochrome display on an IoT device, with a mobile data cap (5MB per month) and would need to update it several times a day. 

A 8bpp bitmap is 32KB (the lowest BmpEncoder can be configured with OOTB), while a 1bpp bitmap is 4KB.

Really hope that this is something that can be done :)

Best regards, Kevin

**Comment by brianpopow** [comment, marked as answer]
The bitmap encoder indeed does only support 8 bit per pixel at the moment. We will need to change the encoder to support 4 bit and 1 bit encoding.

**Reply by JimBobSquarePants** [reply]
@kevinstampe

You managed to nerdsnipe @brianpopow and he rocked out the support over the last two days!!

You can now  pull this build from our MyGet feed which gives you full support. 😄 

https://www.myget.org/feed/sixlabors/package/nuget/SixLabors.ImageSharp/1.0.4-alpha.0.15

**Reply by kevinstampe** [reply]
@brianpopow you've earned yourself a new fan!!!

**Comment by kevinstampe** [comment]
Makes me sad :(
Anyway to pull pixel data out of the image?
I could just avoid using the save command, if i could byte % 255

**Reply by brianpopow** [reply]
To get the pixel data you can use `Image.Load<L8>`, which will load the pixels as `L8` which is one byte per pixel (This is the smallest pixel type we have currently). Then you could use either `GetPixelRowSpan` or `TryGetSinglePixelSpan<L8>` and create a bit array manually from that.
